mne/decoding/transformer.py

```diff
@@ -206,24 +213,48 @@ def transform(self, epochs_data, y=None):
         -------
         X : array, shape (n_epochs, n_channels * n_times)
             The data concatenated over channels
+        y : array, shape(n_epochs,)
+            The label for each epoch.
         """
-        if not isinstance(epochs_data, np.ndarray):
-            raise ValueError("epochs_data should be of type ndarray (got %s)."
-                             % type(epochs_data))
+        from mne import Epochs
 
-        epochs_data = np.atleast_3d(epochs_data)
+        if not isinstance(epochs, Epochs):
+            raise ValueError("epochs should be an instance of Epochs (got %s)"
+                             % type(epochs))
 
-        n_epochs, n_channels, n_times = epochs_data.shape
-        X = epochs_data.reshape(n_epochs, n_channels * n_times)
+        self.epochs_data = np.atleast_3d(self.epochs_data)
+
+        n_epochs, n_channels, n_times = self.epochs_data.shape
+        X = self.epochs_data.reshape(n_epochs, n_channels * n_times)
         # save attributes for inverse_transform
         self.n_epochs = n_epochs
         self.n_channels = n_channels
```

Comment:
save `epochs.info` instead, since the channels need to be described as `mag`, `grad` etc.

Also we'll need many other information such as the `sfreq` to reconstruct the epochs